source/TimePicker.js

```diff
@@ -585,14 +608,27 @@
 		meridiemPickerChanged: function (sender, event) {
 			if(this.syncingPickers) return true;
 
-			var hour = this.$.hour.get('value'),
-				value = event.value;
+			var meridiems = this.meridiems,
+				hour = this.$.hour.getValue(),
+				oldIndex = event.old,
+				newIndex = this.$.meridiem.getValue(),
+				offset, start, end, newHour;
 
-			if (this.value) {
-				// value is 0 for am, 1 for pm
-				// reset the hour to < 12 and then add 12 if it's pm
-				hour = hour%12 + value*12;
-				this.updateHours(hour);
+			offset = hour - parseInt(meridiems[oldIndex]['start'], 10);
```

Comment:
For the Ethiopian time system, you cannot switch the meridiem arbitrarily without updating the time. The system goes like this: 

````
12:00-5:59 morning
6:00       noon
6:01-11:59 afternoon
12:00-5:59 evening
6:00-11:59 night
````

So if the UI is showing 7:00 in the afternoon, and you change the meridiem to evening, I would expect that the time gets updated to 12:00, which is the earliest time in the evening.